pkg/util/validation/validate.go

```diff
@@ -100,6 +120,38 @@ func ValidateSample(cfg SampleValidationConfig, userID string, ls []cortexpb.Lab
 	return nil
 }
 
+func ValidateExemplar(userID string, ls []cortexpb.LabelAdapter, e cortexpb.Exemplar) ValidationError {
```

Comment:
We're very sensitive to performances in the error path. We had issues, reasons why we defined custom errors returned by `ValidateSample` on order to avoid any `fmt.Errorf()` until the `error.Error()` is actually called (we call it only for the 1st error).

Can you do the same here, please?

Reply:
Done, let me know if this looks right to you.

Reply:
Probably should also defer the formatting of the labels (`cortexpb.FromLabelAdaptersToLabels(ls).String()`) for performance as well.